Remotion/Data/DomainObjects.RdbmsTools/RdbmsToolsRunner.cs

```diff
@@ -30,13 +29,32 @@
 using Remotion.Data.DomainObjects.Queries.Configuration;
 using Remotion.Logging;
 using Remotion.Tools;
+#if !NET
+using System.IO;
+#endif
 
 namespace Remotion.Data.DomainObjects.RdbmsTools
 {
   /// <summary>
   /// The <see cref="RdbmsToolsRunner"/> type contains the encapsulates the execution of the various functionality provided by the 
   /// <b>Remotion.Data.DomainObjects.RdbmsTools</b> assembly.
   /// </summary>
+#if NET
+  public class RdbmsToolsRunner : ToolRunner
+  {
+    public static RdbmsToolsRunner Create (RdbmsToolsParameters rdbmsToolsParameters)
+    {
+      return new RdbmsToolsRunner(rdbmsToolsParameters);
+    }
```

Comment:
From the review-perspective, it feels cleaner if we just cut the AppDomain-Stuff.

```suggestion
    public static RdbmsToolsRunner Create (RdbmsToolsParameters rdbmsToolsParameters)
    {
#if NETFRAMEWORK
    AppDomainSetup appDomainSetup = CreateAppDomainSetup(rdbmsToolsParameters);
#endif
      return new RdbmsToolsRunner(
#if NETFRAMEWORK
          appDomainSetup ,
#endif
        rdbmsToolsParameters);
    }
```

I know this is more code but it does make it easier to keep the APIs consistent. Same goes for the rest of the code.

Reply:
The thing here is that there are three things that need to be conditionally switched:

 - The base class
 - The `Create` method
 - The constructor

Right now we have one conditional block that contains all three, if we do the suggested change we would need additional conditional bocks above and below. I don't know if this is easier to read.

Reply:
I get that. it becomes easier to read since R# makes the unsed code gray and in .NET Framework you can just ignore the IF-statements. the advantage is that you only define the API once and some of the smaller duplications (e.g. the field) also get removed. To me, it's just really easy to miss the seocnd part and fixing something when there's a duplicate block. I wouldn't try to prevent code duplication on code that's typically not written in separate lines, but here we only need to guard a couple of lines of code that may well be written in separate lines, e.g. the parameter list.

Whether or not to actually duplicate the ctor call or just cut out the parameter, yeah, that's up to a coin toss. The rest though, the cut is made with the intention of creating the most contiguous conditional block possible which, too me, feels like the not so precise choice. For instance, when we place the contional all around the factory method, the factory method itself becomes redundant for .NET. I'm not saying that we need to keep it in the future, though at this time, it's best to keep it so the APIs on the callsite remain unchanged. When we put this all in a big conditional block, such reasons become less obvious and the entire block is suddenly open to big refactorings even though that wasn't the intent of the .NET/.NETFRAMEWORK blocks.